ambari-server/src/main/resources/stacks/BIGTOP/3.3.0/services/RANGER/package/scripts/setup_ranger_xml.py

```diff
@@ -465,24 +465,43 @@ def password_validation(password):
 def copy_jdbc_connector(ranger_home):
   import params
 
-  if params.jdbc_jar_name is None and params.driver_curl_source.endswith("/None"):
+  if params.jdbc_jar_name is None or params.driver_curl_source.endswith("/None"):
     error_message = format(
       "{db_flavor} jdbc driver cannot be downloaded from {jdk_location}\nPlease run 'ambari-server setup --jdbc-db={db_flavor} --jdbc-driver={{path_to_jdbc}}' on ambari-server host."
     )
     raise Fail(error_message)
 
-  if params.driver_curl_source and not params.driver_curl_source.endswith("/None"):
-    if params.previous_jdbc_jar and os.path.isfile(params.previous_jdbc_jar):
-      File(params.previous_jdbc_jar, action="delete")
+  ModifyPropertiesFile(
+    format("{ranger_home}/install.properties"),
+    properties=params.config["configurations"]["admin-properties"],
+    owner=params.unix_user,
+  )
+
+  if params.db_flavor.lower() == "sqla":
+    ModifyPropertiesFile(
+      format("{ranger_home}/install.properties"),
+      properties={"SQL_CONNECTOR_JAR": format("{ranger_home}/ews/lib/sajdbc4.jar")},
+      owner=params.unix_user,
+    )
+  else:
+    ModifyPropertiesFile(
+      format("{ranger_home}/install.properties"),
+      properties={"SQL_CONNECTOR_JAR": format("{driver_curl_target}")},
+      owner=params.unix_user,
+    )
```

Comment:
The intended configuration should be applied regardless of whether the JDBC jar has been already downloaded or not.